File: library/library-client/src/app/core/models/books/Book.ts
import { Author } from '../authors/Author';
import { Category } from '../categories/Category';
import { FavoriteBook } from '../favoriteBook/FavoriteBook';

export type Book = {
  id: number;
  title: string;
  pixUrl: string;
  authorId?: number;
  categoryId?: number;
  publicationDate?: Date;
  author?: Author;
  category?: Category;
  favorite?: FavoriteBook;
};

File: library/library-api/LibraryApi/Infrastructure/Models/Book/Book.cs
namespace LibraryApi.Infrastructure.Models.Book;

public class Book
{
    public int Id { get; set; }
    public DateTime? PublicationDate { get; set; }
    public string Title { get; set; }
    public string PixUrl { get; set; }
    public int? AuthorId { get; set; }
    public int? CategoryId { get; set; }
}